src/launcher/java/org/truffleruby/launcher/options/CommandLineParser.java

```diff
@@ -450,6 +465,7 @@ private void processArgument() throws CommandLineException {
                             // ruby interpreter compatibilty
                             // Usage: ruby [switches] [--] [programfile] [arguments])
                             endOfInterpreterArguments = true;
+                            lastInterpreterArgumentIndex = argumentIndex;
```

Comment:
Does `endOfInterpreterArguments == (lastInterpreterArgumentIndex != -1)` hold?
If so we could have only `lastInterpreterArgumentIndex`.

Reply:
yes, I wanted to keep the Boolean to make it easier to understand. However I can turn it into `endOfInterpreterArguments() {lastInterpreterArgumentIndex != -1}` which will be ok as well.

Reply:
That sounds good, it makes it easier to not forget to update one of them.